src/js/controller/instream-html5.js

```diff
@@ -54,25 +56,22 @@ const InstreamHtml5 = function(_controller, _model) {
         mediaElement.addEventListener('abort', _this.srcReset);
         mediaElement.addEventListener('emptied', _this.srcReset);
         this._adModel = _adModel;
+        _programController = new ProgramController(_adModel);
     };
 
     /** Load an instream item and initialize playback **/
-    _this.load = function() {
+    _this.load = function(item) {
         // Let the player media model know we're using it's video tag
         _this.srcReset();
 
         // Make sure it chooses a provider
-        _adModel.stopVideo();
-        _adModel.setItemIndex(0).then(() => {
-            if (!_adModel) {
-                return;
-            }
-            _checkProvider(_adModel.getVideo());
-        });
+        _programController.stopVideo();
         _checkProvider();
```

Comment:
@johnBartos is it still necessary to checkProvider here ? Shouldn't this be moved to program controller or its dependencies ?

Reply:
Seems like a lot of bespoke ads logic that shouldn't be mixed with the program controller. Instream-html5 will change when doing the BGL ticket itself